### PR TITLE
[StructuralMechanicsApplication] Clean up surface load

### PR DESCRIPTION
**Pull request by loumalouomega**
While looking for the error in #3578 I cleaned up the surface load. This is just that, a clean up. I added three very basic tests more

### CODE REVIEW

**Comment by loumalouomega** [comment]
Ping

File: applications/StructuralMechanicsApplication/custom_conditions/surface_load_condition_3d.cpp [hunk 1]
```diff
@@ -12,320 +12,291 @@
...
+    ProcessInfo& rCurrentProcessInfo,
+    const bool CalculateStiffnessMatrixFlag,
+    const bool CalculateResidualVectorFlag
+    )
+{
+    KRATOS_TRY;
 
-            rRightHandSideVector = ZeroVector(mat_size); //resetting RHS
-        }
+    auto& r_geometry = GetGeometry();
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_conditions/surface_load_condition_3d.cpp]: const

File: applications/StructuralMechanicsApplication/custom_conditions/surface_load_condition_3d.cpp [hunk 2]
```diff
@@ -12,320 +12,291 @@
...
-            }
+    array_1d<double, 3 > tangent_xi, tangent_eta;
+
+    // Iterate over the Gauss points
+    Matrix J(3, 2);
+    for (std::size_t point_number = 0; point_number < integration_points.size(); ++point_number) {
+        r_geometry.Jacobian(J, point_number, integration_method);
+        const double detJ = MathUtils<double>::GeneralizedDet(J);
+        const double integration_weight = GetIntegrationWeight(integration_points, point_number, detJ);
+        const auto& N = row(Ncontainer, point_number);
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_conditions/surface_load_condition_3d.cpp]: rN

File: applications/StructuralMechanicsApplication/custom_conditions/surface_load_condition_3d.cpp [hunk 3]
```diff
@@ -12,320 +12,291 @@
...
+    }
+
+    KRATOS_CATCH("")
+}
+
+//***********************************************************************************
+//***********************************************************************************
+
+void SurfaceLoadCondition3D::MakeCrossMatrix(
+    BoundedMatrix<double, 3, 3>& M,
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_conditions/surface_load_condition_3d.cpp]: rM

File: applications/StructuralMechanicsApplication/custom_conditions/surface_load_condition_3d.cpp [hunk 4]
```diff
@@ -12,320 +12,291 @@
...
+    const array_1d<double, 3>& U
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_conditions/surface_load_condition_3d.cpp]: rU

**Review by philbucher — COMMENTED** [review]
some minor comments then I will approve

**Comment by loumalouomega** [comment]
Done @philbucher, I added extra Doxygen commentaries

**Review by philbucher — APPROVED** [review]
thanks for cleaning up

**Pull request merged**